File: app/src/main/java/org/publicaid/app/data/model/ApiResponse.kt
package org.publicaid.app.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/** Standard API success envelope. */
@JsonClass(generateAdapter = true)
data class ApiResponse<T>(
    val data: T,
    val meta: ApiMeta? = null,
)

@JsonClass(generateAdapter = true)
data class ApiMeta(
    val total: Int? = null,
    val limit: Int? = null,
    val offset: Int? = null,
    @Json(name = "took_ms") val tookMs: Long = 0,
)

/** Standard API error envelope. */
@JsonClass(generateAdapter = true)
data class ApiErrorResponse(
    val error: ApiErrorDetail,
)

@JsonClass(generateAdapter = true)
data class ApiErrorDetail(
    val code: String,
    val message: String,
)

@JsonClass(generateAdapter = true)
data class StatsResponse(
    @Json(name = "total_entities") val totalEntities: Int,
    @Json(name = "total_categories") val totalCategories: Int,
)

File: lib/models/api_response.dart
class ApiResponse<T> {
  final T data;
  final ApiMeta? meta;

  ApiResponse({required this.data, this.meta});
}

class ApiMeta {
  final int total;
  final int limit;
  final int offset;

  ApiMeta({required this.total, required this.limit, required this.offset});

  factory ApiMeta.fromJson(Map<String, dynamic> json) {
    return ApiMeta(
      total: json['total'] as int? ?? 0,
      limit: json['limit'] as int? ?? 20,
      offset: json['offset'] as int? ?? 0,
    );
  }
}
